src/sentry/silo/client.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+from typing import List
+
+from django.conf import settings
+
+from sentry.shared_integrations.client.base import BaseApiClient
+from sentry.silo.base import SiloMode
+from sentry.types.region import get_region_by_name
+
+
+class SiloClientError(Exception):
+    """Indicates an error in processing a cross-silo HTTP request"""
+
+
+class BaseSiloClient(BaseApiClient):
+    integration_type = "silo_client"
+
+    @property
+    def access_modes() -> List[SiloMode]:
+        """
+        Limit access to the client to only the SiloModes set here.
+        """
+        raise NotImplementedError
+
+    def create_base_url(self, address: str) -> str:
+        # We're intentionally not exposing API Versioning as an interface just yet
+        # This may be revisited once more of deployment drift in Hybrid Cloud is pinned down.
+        api_prefix = "/api/0"
+        return f"{address}{api_prefix}"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if SiloMode.get_current_mode() not in self.access_modes:
+            access_mode_str = ", ".join(str(m) for m in self.access_modes)
+            raise SiloClientError(
+                f"Cannot invoke {self.__class__.__name__} from {SiloMode.get_current_mode()}. "
+                f"Only available in: {access_mode_str}"
+            )
+
+    def request(self, *args, **kwargs):
+        # TODO: Establish a scheme to authorize requests across silos
+        # (e.g. signing secrets, JWTs)
+        super().request(*args, kwargs)
+        # TODO: Establish a scheme to check/log the Sentry Version of the requestor and server
+        # optionally raising an error to alert developers of version drift
+
+
+class RegionSiloClient(BaseSiloClient):
+    access_modes = [SiloMode.CONTROL]
+
+    datadog_prefix = "silo_client.region"
```

Comment:
We probably shouldn't be calling this `datadog_prefix` but `metrics_prefix`. I know that means changing integrations as that is where this property is being defined.

Reply:
Yep, I agree. Same goes for `integration_type` property to something more generic like `client_type` since it's not just for integrations anymore.

Reply:
I'll make these changes in a follow up PR!